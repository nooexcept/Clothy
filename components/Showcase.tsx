import * as React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'

const useStyles = makeStyles({
    root: {
        width: '30vw',
        minWidth: 200,
        background: '#000',
    },
    media: {
        height: '55vh',
        minHeight: 150,
    },
    content: {
        backgroundColor: '#fbd46d',
    },
    actions: {
        backgroundColor: '#fbd46d',
    },
})

interface ShowcaseProps {
    img: string
    title: string
    description: string
}

export default function Showcase(props: ShowcaseProps): React.ReactElement {
    const classes = useStyles()

    return (
        <Card className={classes.root}>
            <CardActionArea>
                <CardMedia
                    className={classes.media}
                    image={props.img}
                    title={props.title}
                />
                <CardContent className={classes.content}>
                    <Typography
                        color="textPrimary"
                        gutterBottom
                        variant="h5"
                        component="h2"
                    >
                        {props.title}
                    </Typography>
                    <Typography
                        variant="body2"
                        color="textPrimary"
                        component="p"
                        noWrap
                    >
                        {props.description}
                    </Typography>
                </CardContent>
            </CardActionArea>
            <CardActions className={classes.actions} disableSpacing>
                <Button size="small">
                    <Typography color="textPrimary" variant="body1">
                        Share
                    </Typography>
                </Button>
                <Button size="small">
                    <Typography color="textPrimary" variant="body1">
                        Browse
                    </Typography>
                </Button>
            </CardActions>
        </Card>
    )
}
