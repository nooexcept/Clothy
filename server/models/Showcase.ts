import mongoose from 'mongoose'

export interface Showcase extends mongoose.Document {
    collId: string
}

const showcaseSchema = new mongoose.Schema({
    collId: {
        type: String,
        required: true,
    },
})

export default mongoose.model<Showcase>('Showcase', showcaseSchema)
