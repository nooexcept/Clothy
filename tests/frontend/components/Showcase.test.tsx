import React from 'react'
import { ShallowWrapper } from 'enzyme'
import { createShallow } from '@material-ui/core/test-utils'
import Showcase from '../../../components/Showcase'

let shallow: ReturnType<typeof createShallow>
let wrapper: ShallowWrapper

const mockedProd = {
    title: 'Ultra chill jumpers',
    description: 'If you wanna look chill, you need ultra chill jumpers',
    img: '/photo-of-woman-standing-on-staircase-3120339.jpg',
}

beforeEach(() => {
    shallow = createShallow()
    wrapper = shallow(<Showcase {...mockedProd} />)
})

describe('<Showcase /> rendering', () => {
    it('should render without problems', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
