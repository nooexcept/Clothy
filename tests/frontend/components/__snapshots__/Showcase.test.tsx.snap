// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Showcase /> rendering should render without problems 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
      image="/photo-of-woman-standing-on-staircase-3120339.jpg"
      title="Ultra chill jumpers"
    />
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-content-3"
    >
      <WithStyles(ForwardRef(Typography))
        color="textPrimary"
        component="h2"
        gutterBottom={true}
        variant="h5"
      >
        Ultra chill jumpers
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textPrimary"
        component="p"
        noWrap={true}
        variant="body2"
      >
        If you wanna look chill, you need ultra chill jumpers
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))
    className="makeStyles-actions-4"
    disableSpacing={true}
  >
    <WithStyles(ForwardRef(Button))
      size="small"
    >
      <WithStyles(ForwardRef(Typography))
        color="textPrimary"
        variant="body1"
      >
        Share
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      size="small"
    >
      <WithStyles(ForwardRef(Typography))
        color="textPrimary"
        variant="body1"
      >
        Browse
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
